src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = tail = new Node(null, value, null);
```

Comment:
When creating a new Node<T>, pass all the fields to the Node constructor. Here, you are passing `null` as the `prev` and `next` parameters. Instead, you should pass the actual previous and next nodes. This will make your code more robust and prevent null pointer exceptions.